### PR TITLE
Boyscout/refactor auth client

### PR DESCRIPTION
**Pull request by NoNamer777**
Refactors Authentication process for the client.

**NOTE: This PR does depend on changes in from [this PR](https://github.com/NoNamer777/WA-2022-group-2/pull/17), so I'll only merge this PR only when that PR is handled.**

- Add services for:
  - ~Centralized access to and from the localstorage using the `StorageService`~
  - Making http requests and handles their responses using the `HttpRequestService`
    - Writes JWT tokens to the localstorage when the `Authorization` header is present on a response
    - Reads the stored JWT token and adds the `Authorization` header to requests when the token is found in the localstorage
  - Making API calls for Users data using the `UserService`
  - Making API calls for authentication/authorization using the `AuthService`
  - Decoding JWT tokens in the client using the `JwtService`
- Refactored the `AuthStore` so that:
  - it uses async/await code style properly
  - it uses the `setup` vue code style instead of the `options` vue code style
  Which enables use to also declare and reuse "private" functions/variables (currently only used for the `populateAuthenticatedUser` function
- Make sure the `AuthStore` is initialized before the rest of the application so that we can use it to make sure that routes are properly guarded
- The `.env` file is moved from the server project to a shared location and environment variables are loaded from there for both the server and client project. The README in the root of the repository has been updated for this change. Please, have a look there and let met know if something's unclear.
  - For environment variables that need to be used in the client project it is important to prefix them with `VITE_` otherwise they won't be usable. https://vitejs.dev/guide/env-and-mode.html#env-files

### CODE REVIEW

File: packages/client/src/app/services/storage.service.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+export class StorageService {
```
Comment (ahmed-ben10) [inline packages/client/src/app/services/storage.service.js:1]: This class adds no functionality and you can just use local storage directly

**Comment by ahmed-ben10** [comment]
Looks good! left a few comments

**Comment by NoNamer777** [comment]
> Looks good! left a few comments

I think your comments are still pending, because I can't see them 🤔

**Review by ahmed-ben10 — CHANGES_REQUESTED** [review]

**Comment by ahmed-ben10** [comment]
Whoops! i forgot to push the button

File: packages/client/src/app/services/auth.service.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  async login(username, password) {
+    return await HttpRequestService.instance().postRequest('/auth/login', { username, password })
+  }
+
+  /**
+   * @param userData {Omit<User, 'id' | 'isAdmin'> & { passwordConfirm: string }}
+   * @return {Promise<User>}
+   */
+  async register(userData) {
+    if (userData.password !== userData.passwordConfirm) {
```
[the remark this answers is not in the document]
Comment (NoNamer777) [reply packages/client/src/app/services/auth.service.js:30]: Ah, guessed I overlooked that. Neat 👌🏼 
Then I'll remove this check

File: packages/client/src/app/services/storage.service.js (hunk 1 above)
Comment (NoNamer777) [reply packages/client/src/app/services/storage.service.js:1]: Removed the StorageService

**Review by ahmed-ben10 — APPROVED** [review]
LGTM :+1:

**Pull request merged**